socialhome/streams/app/components/Stream.vue

```diff
@@ -82,19 +94,30 @@ export default Vue.component("stream", {
         onNewContentClick() {
             this.$store.dispatch(streamStoreOperations.newContentAck)
         },
+        loadStream(lastId = undefined) {
+            let options = lastId ? {params: {lastId}} : {params: {}}
+
+            if (this.$store.state.streamName.match(/^followed/)) {
+                this.$store.dispatch(streamStoreOperations.getFollowedStream, options)
+            } else if (this.$store.state.streamName.match(/^public/)) {
+                this.$store.dispatch(streamStoreOperations.getPublicStream, options)
+            } else if (this.$store.state.streamName.match(/^tag/)) {
+                options.params.name = this.$store.state.tagName
+                this.$store.dispatch(streamStoreOperations.getTagStream, options)
+            } else if (this.$store.state.streamName.match(/^profile_all/)) {
+                options.params.id = this.currentBrowsingProfileId
+                this.$store.dispatch(streamStoreOperations.getProfileAll, options)
+            } else if (this.$store.state.streamName.match(/^profile_pinned/)) {
+                options.params.id = this.currentBrowsingProfileId
+                this.$store.dispatch(streamStoreOperations.getProfilePinned, options)
+            }
+        },
+        loadMore() {
+            this.loadStream(this.$store.state.contentIds[this.$store.state.contentIds.length - 1])
```

Comment:
This looks wrong. It should the *smallest* `throughIds`, not the last element in `contentIds`. You can use something like [this](https://github.com/jaywink/socialhome/blob/7d3b0bd0b93e5580706686a38e6d3987767e0d2a/socialhome/static/js/streams.js#L116).

Last element doesn't mean smallest. It might in practise, but it's not a good idea to rely on the order. We will pop stuff into the list on top of the stream, for example.

Reply:
Ah, yes, indeed.